MvvmCross/IoC/MvxTypeExtensions.cs

```diff
@@ -239,5 +239,54 @@ public static object CreateDefault(this Type type)
 
             return Activator.CreateInstance(type);
         }
+        
+        public static ConstructorInfo FindApplicableConstructor(this Type type, IDictionary<string, object> arguments)
+        {
+            var constructors = type.GetConstructors();
+            if (arguments == null || arguments.Count == 0)
+            {
+                return constructors.FirstOrDefault();
+            }
+            
+            var names = arguments.Keys;
+
+            foreach (var constructor in constructors)
+            {
+                var parameters = constructor.GetParameters();
+                if (parameters.All(p => names.Contains(p.Name)))
+                {
+                    return constructor;
+                }
+            }
+
+            return null;
+        }
+        
+        public static ConstructorInfo FindApplicableConstructor(this Type type, object[] arguments)
+        {
+            var constructors = type.GetConstructors();
+            
+            foreach (var constructor in constructors)
+            {
+                var parameterTypes = constructor.GetParameters().Select(p => p.ParameterType);
+                var unusedArguments = arguments.ToList();
```

Comment:
We should check here to see if number of arguments matches the number of parameters to the constructor. However, also need to consider optional parameters

Reply:
That's what I wanted to change. Before, `IocConstruct` could (in some cases, not all) only inject the arguments that were supplied. But why would you use it then? With these changes, you can have some params supplied by args and some others injected from registrations. So it's possible to have more parameters than arguments. What this checks is that all the supplied arguments can be used in the constructor, otherwise you probably meant to use a different constructor.